src/components/CategoryFilter.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import categories from '../data/categories';
+
+function CategoryFilter({ setFilter, filter }) {
+  function handleFilterChange(e) {
+    setFilter(e.target.value);
+  }
```

Comment:
According to the requirements, you need to have the handler function on `BookList.js` and leave this just as a presentational component. Instead of  `setFilter` it should receive the handler function. So the component should basicly become something like this:

```js
const options = ["ALL", ...categories].map(x => (
  <option key={x} value={x}>
    {x}
  </option>
));

const CategoryFilter = ({ handleFilterChange, filter }) => (
  <select
    id="categoryFilter"
    onChange={handleFilterChange}
    defaultValue={filter}
  >
    {options}
  </select>
);
```